crates/store/src/db/mod.rs

```diff
@@ -93,11 +93,43 @@ pub struct StateSyncUpdate {
 
 #[derive(Debug, PartialEq)]
 pub struct NoteSyncUpdate {
-    pub notes: Vec<NoteRecord>,
+    pub notes: Vec<NoteSyncRecord>,
     pub block_header: BlockHeader,
     pub chain_tip: BlockNumber,
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct NoteSyncRecord {
+    pub block_num: BlockNumber,
+    pub note_index: BlockNoteIndex,
+    pub note_id: RpoDigest,
+    pub metadata: NoteMetadata,
+    pub merkle_path: MerklePath,
+}
+
```

Comment:
Consider moving this to the `proto` crate

Reply:
I was thinking about it and decided to put this struct to the store because it is such an adhoc structure by its nature, but in a case of moving it to proto we would have to implement backward conversion (from proto to domain type) and it's not enough fields in proto to do it (but we can implement something like `impl From<(BlockNumber, NoteSyncRecord)> for NoteSyncRecordPb`).